packages/stellar-axelar-std-derive/src/contractimpl.rs

```diff
@@ -35,6 +44,57 @@ pub fn contractimpl(impl_block: &mut ItemImpl) -> Result<proc_macro2::TokenStrea
     })
 }
 
+fn all_contract_endpoints(
+    impl_block: &ItemImpl,
+) -> impl Fn(&mut ImplItem) -> Option<&mut ImplItemFn> {
+    let any_contract_endpoint = if impl_block.trait_.is_some() {
+        any
+    } else {
+        any_pub_fn
+    };
+
+    move |item| {
+        any_fn(item)
+            .and_then(any_contract_endpoint)
+            .filter(has_env_param)
+    }
+}
+
+fn instance_ttl_extension(method: &mut ImplItemFn) -> Result<(), syn::Error> {
+    let env_ident = parse_env_identifier(&method.sig.inputs)?;
+
+    let extend_ttl_stmt: Stmt = if is_env_ref(method, env_ident) {
+        parse_quote! {
+            stellar_axelar_std::ttl::extend_instance_ttl(#env_ident);
+        }
+    } else {
```

Comment:
no need for this, the compiler automatically derefs multiple references, i.e. `&&env` will get resolved to `&env` anyway